### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by aabdoo23**
#### PR Classification
New feature and code structure enhancement.

#### PR Summary
Introduced Entity Framework Core packages and restructured project files to improve service registration and repository patterns.
- Added Entity Framework Core packages to multiple project files.
- Updated `Readme.txt` to specify folder structure for entities.
- Created `ConfigureServices` class with `AddServices` method in `SchedsForums.Domain.Common`.
- Introduced `IBaseEntity` and `IBaseRepository<T>` interfaces in `SchedsForums.Domain.Interfaces`.
- Added folder structures for `Entities`, `Exceptions`, `Services`, `Repositories`, `Contexts`, and `Migrations` in several project files.

### CODE REVIEW

File: SchedsForums.Domain/Common/ConfigureServices.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SchedsForums.Domain.Common
```
Comment (MostafaHgz) [inline SchedsForums.Domain/Common/ConfigureServices.cs:3]: remove this file

File: SchedsForums.Domain/Interfaces/IBaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+namespace SchedsForums.Domain.Interfaces
+{
+    public interface IBaseRepository<T> where T:IBaseEntity
+    {
+        Task<T> GetByIdAsync(int id);
+        Task<IEnumerable<T>> GetAllAsync();
+        Task<T> AddAsync(T entity);
+        Task<T> UpdateAsync(T entity);
+        Task<T> DeleteAsync(int id);
+
```
Comment (MostafaHgz) [inline SchedsForums.Domain/Interfaces/IBaseRepository.cs:16]: remove this extra line

File: SchedsForums.Api/SchedsForums.Api.csproj [hunk 3]
```diff
@@ -7,6 +7,16 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.10" />
```
Comment (MostafaHgz) [inline SchedsForums.Api/SchedsForums.Api.csproj:10]: you didn't add the db connection yet. please add it

**Review by MostafaHgz — CHANGES_REQUESTED** [review]

File: SchedsForums.Api/SchedsForums.Api.csproj (hunk 3 above)
Comment (aabdoo23) [reply SchedsForums.Api/SchedsForums.Api.csproj:10]: done

File: SchedsForums.Domain/Common/ConfigureServices.cs (hunk 1 above)
Comment (aabdoo23) [reply SchedsForums.Domain/Common/ConfigureServices.cs:3]: leh, I created it later to register all the services here instead of crowding the program.cs

**Pull request closed without merging**